src/components/feedback/feedback.js

```diff
@@ -1,321 +1,222 @@
-                )}
-              </div>
-
-              <div className="txt-s txt-em">
-                This form is for documentation feedback. If you have a technical
-                question about how to use a Mapbox product,{' '}
-                <a href="https://support.mapbox.com/hc/en-us" className="link">
-                  contact Support
-                </a>
-                .
               </div>
-            </div>
-          )}
+            )}
+          </div>
+        </div>
       </div>
     );
   }
-}
 
-// character counter that appears in the bottom-right of the feedback textarea
-class FeedbackCounter extends React.PureComponent {
   render() {
-    return (
-      <div
-        id={this.props.createId('counter')}
-        className={classnames(
-          'absolute bottom right mb6 mr18 txt-mono bg-lighten75 px3 txt-s',
-          {
-            'color-red': this.props.feedbackOverLimit
-          }
-        )}
-      >
-        {this.props.feedbackLength}
-      </div>
-    );
-  }
-}
+    const { isOpen, category, sentFeedback } = this.state;
+    const { type } = this.props;
 
-FeedbackCounter.propTypes = {
-  createId: PropTypes.func.isRequired,
-  feedbackOverLimit: PropTypes.bool.isRequired,
-  feedbackLength: PropTypes.number.isRequired
-};
+    const FeedbackWrapper = (props) => this.renderWrapper(props);
 
-// inline warning message that will appear if the user enters > 1000 characters in the feedback textarea
-class FeedbackOverlimitWarning extends React.PureComponent {
-  render() {
+    /** This component returns early to improve code clarity **/
+
+    // Initial stage: the user has not clicked on feedback
+    if (!isOpen) {
+      return (
+        <button value="Share" onClick={this.openFeedback} className="btn">
+          Share your feedback
+        </button>
+      );
+    }
+    // Second stage: select the category
+    if (isOpen && !category) {
+      return (
+        <FeedbackWrapper
+          title={`Share your feedback${
+            type !== 'page' ? ` for this ${type}` : ''
+          }`}
+        >
+          {Object.keys(this.categories()).map((category) => (
+            <button
+              value={category}
+              onClick={this.selectCategory}
+              className="btn w-full mb6"
+              key={category}
+            >
+              {category}
+            </button>
+          ))}
+        </FeedbackWrapper>
+      );
+    }
+    // Final stage: a confirmation after the user sends feedback
+    if (isOpen && category && sentFeedback) {
+      return (
+        <FeedbackWrapper>
+          <div className="align-center prose relative">
+            <div className="mt-neg30 inline-block">
+              <BookImage size={75} />
+            </div>
+            <p>
+              <strong>Thank you!</strong>
+            </p>
+            <p>
+              Our documentation team will read your feedback. Thank you for
+              helping us improve this page.
```

Comment:
Should this `page` be a variable, too? Or is that overly complicated for something that won't add any clarity to the feedback submitting process?

Reply:
Ohhh I think we can variabl-ize it!